apps/emqx/src/emqx_channel.erl

```diff
@@ -1441,22 +1433,21 @@ check_pub_caps(#mqtt_packet{header = #mqtt_packet_header{qos    = QoS,
 check_sub_acls(TopicFilters, Channel) ->
     check_sub_acls(TopicFilters, Channel, []).
 
-check_sub_acls([ TopicFilter = {Topic, _} | More] , Channel, Acc) ->
-    case check_sub_acl(Topic, Channel) of
+check_sub_acls([ TopicFilter = {Topic, _} | More] , Channel = #channel{clientinfo = ClientInfo}, Acc) ->
+    case emqx_access_control:authorize(ClientInfo, subscribe, Topic) of
```

Comment:
It is determined by emqx_access_control whether to disconnect unauthorized clients. Is this supposed to be a bad practice?

Would it be better to leave it to the caller itself to decide if it needs to disconnect?